control-plane/agents/src/bin/core/controller/reconciler/volume/garbage_collector.rs

```diff
@@ -345,62 +338,42 @@ async fn disown_non_reservable_replicas(
     let mut results = vec![];
 
     // Fetch all nexuses that are not properly shutdown
-    let shutdown_failed_nexuses: Vec<ResourceMutex<NexusSpec>> = context
+    let shutdown_failed_nexuses = context
         .registry()
         .specs()
         .volume_failed_shutdown_nexuses(volume.uuid())
         .await;
 
-    // Remove the local child from the shutdown pending nexuses as they are
-    // non reservable.
+    // Remove the local children from the volume as they are non-reservable.
     for nexus in shutdown_failed_nexuses {
+        if target.uuid() == nexus.uuid() {
```

Comment:
IIUC `target` represents current nexus here? If that has failed shutdown, why don't we want to go down the same code path for such target?

Reply:
In this case we probably don't want to be disowning children outside of the existing logic for managing the volume target, as it may interfere with existing logic